components/ontology.js

```diff
@@ -0,0 +1,45 @@
+// node_modules
+import PropTypes from "prop-types"
+
+/**
+ * Map from ontology term ID prefixes to corresponding URL bases.
+ */
+const urlMap = {
+  EFO: "http://www.ebi.ac.uk/efo/",
+  UBERON:
+    "http://www.ontobee.org/ontology/UBERON?iri=http://purl.obolibrary.org/obo/",
+  CL: "http://www.ontobee.org/ontology/CL?iri=http://purl.obolibrary.org/obo/",
+  CLO: "http://www.ontobee.org/ontology/CLO?iri=http://purl.obolibrary.org/obo/",
+}
+
+/**
+ * Display the ontology term ID given in `termId`, and link to a corresponding site if the prefix
+ * of the term ID has one. Any term IDs with prefixes not matching any in the `urlMap` global
+ * variable simply display without a link. All term IDs are of the form XXX:nnnnnnn...
+ */
+export const OntologyTermId = ({ termId }) => {
```

Comment:
It looks like the only place we get an ontology term CURIE is from `{ontologyTerm.term_id}` above. How is the `ontologyTerm` defined? Especially since these are ontology terms and not instance IDs (like a database ID: `FB:12335`) and if we knew how these `term_id`s are assembled we could nearly guarantee they have only one colon in the ID. Additionally, one strategy when dealing with CURIEs I've used before is to parse them into the "namespace" and "local ID" part, so you just have an object representing a correct ID. Maybe this compound object could be recieved by OntologyTermId instead, simplifying the checks?

Reply:
@dougli1sqrd
> parse them into the "namespace" and "local ID" part, so you just have an object representing a correct ID
You mean anywhere we deal with an ontology term, pass around an object like:
```
{
  namespace: "EFO",
  id: "123"
}
```
?

Reply:
…or even a class if that’s not overkill.

Reply:
Yeah something like that!